presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -129,6 +126,25 @@ private SubPlan buildFragment(PlanNode root, FragmentProperties properties, Plan
             return new SubPlan(fragment, properties.getChildren());
         }
 
+        private static Set<Symbol> extractSymbols(PlanNode planNode)
```

Comment:
I would prefer to organize symbol extracting into well defined methods (e.g: `extractDepdencies`, `ex, see: https://github.com/prestodb/presto/issues/8552. It's not clean what `extractSymbols` does since there away various ways of extracting symbols.

This looks like utility method